WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsFormatDialog.java

```diff
@@ -0,0 +1,189 @@
+import android.app.Dialog;
+import android.app.DialogFragment;
+import android.app.Fragment;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.StringRes;
+import android.text.TextUtils;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.LinearLayout;
+import android.widget.RadioButton;
+import android.widget.RadioGroup;
+import android.widget.TextView;
+
+import org.wordpress.android.Constants;
+import org.wordpress.android.R;
+import org.wordpress.android.ui.ActivityLauncher;
+import org.wordpress.android.util.EditTextUtils;
+
+/**
+ * Custom date/time format dialog
+ */
+
+public class SiteSettingsFormatDialog extends DialogFragment implements DialogInterface.OnClickListener {
+
+    public enum FormatType {
+        DATE_FORMAT,
+        TIME_FORMAT;
+
+        public String[] getEntries(@NonNull Context context) {
+            if (this == FormatType.DATE_FORMAT) {
+                return context.getResources().getStringArray(R.array.date_format_entries);
+            } else {
+                return context.getResources().getStringArray(R.array.time_format_entries);
+            }
+        }
+
+        public String[] getValues(@NonNull Context context) {
+            if (this == FormatType.DATE_FORMAT) {
+                return context.getResources().getStringArray(R.array.date_format_values);
+            } else {
+                return context.getResources().getStringArray(R.array.time_format_values);
+            }
+        }
+    }
+
+    private static final String KEY_FORMAT_TYPE = "format_type";
+    public static final String KEY_FORMAT_VALUE = "format_value";
+
+    private String mFormatValue;
+    private boolean mConfirmed;
+
+    private EditText mEditCustomFormat;
+    private RadioGroup mRadioGroup;
+
+    private String[] mEntries;
+    private String[] mValues;
+
+    public static SiteSettingsFormatDialog newInstance(@NonNull FormatType formatType, @NonNull String formatValue) {
+        Bundle args = new Bundle();
+        args.putSerializable(KEY_FORMAT_TYPE, formatType);
+        args.putString(KEY_FORMAT_VALUE, formatValue);
+
+        SiteSettingsFormatDialog dialog = new SiteSettingsFormatDialog();
+        dialog.setArguments(args);
+        return dialog;
+    }
+
+    @Override
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+        View view = View.inflate(getActivity(), R.layout.site_settings_format_dialog, null);
+        TextView txtTitle = view.findViewById(R.id.text_title);
+        TextView txtHelp = view.findViewById(R.id.text_help);
+        mEditCustomFormat = view.findViewById(R.id.edit_custom);
+        mRadioGroup = view.findViewById(R.id.radio_group);
+
+        Bundle args = getArguments();
+        FormatType formatType = (FormatType) args.getSerializable(KEY_FORMAT_TYPE);
```

Comment:
Although it shouldn't normally happen, AS complains about a possible NPE for the usage of `formatType` later on since this doesn't guarantee that it'll be set here. How do you feel about adding a fallback for that, something like below:

```
if (formatType == null) {
    formatType = FormatType.DATE_FORMAT;
}
```

Even if it's not actually used, it'll at least make AS happy. What do you think?